internal/secrets/drivertest/drivertest.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2018 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package drivertest provides a conformance test for implementations of
+// driver.
+package drivertest
+
+import (
+	"context"
+	"github.com/google/go-cloud/internal/secrets/driver"
+	"github.com/google/go-cmp/cmp"
+	"testing"
+)
+
+// Harness descibes the functionality test harnesses must provide to run
+// conformance tests.
+type Harness interface {
+	// MakeDriver returns an Encypter and a Decrypter for use in tests.
+	MakeDriver(ctx context.Context) (driver.Encrypter, driver.Decrypter, error)
+}
+
+// HarnessMaker describes functions that construct a harness for running tests.
+// It is called exactly once per test.
+type HarnessMaker func(ctx context.Context, t *testing.T) (Harness, error)
+
+// RunConformanceTests runs conformance tests for provider implementations of secret management.
+func RunConformanceTests(t *testing.T, newHarness HarnessMaker) {
+	t.Run("TestEncryptDecrypt", func(t *testing.T) {
+		testEncryptDecrypt(t, newHarness)
+	})
+}
+
+// testEncryptDecrypt tests the functionality of encryption and decryption
```

Comment:
Things to consider for future conformance tests (maybe add TODOs and/or file an issue):

- Verify that encrypting the same message with the same key twice doesn't result in the same encrypted bytes.
- Verify that different keys work (and again, don't result in the same encrypted bytes).

Reply:
I can add those to this PR or do TODOs/issue, which is preferred?

Reply:
I'd vote for TODOs/issue, whichever you prefer; the additional tests are not critical for this PR and it's easier/better to keep the PRs small and focused.

Reply:
issue added: https://github.com/google/go-cloud/issues/919